pinot-query-planner/src/main/java/org/apache/pinot/query/QueryEnvironment.java

```diff
@@ -132,6 +134,13 @@ public QueryEnvironment(TypeFactory typeFactory, CalciteSchema rootSchema, Worke
     _hepProgram = hepProgramBuilder.build();
   }
 
+  /**
+   * Returns the RelRoot generated after compiling the query.
+   */
+  public RelRoot getRelRoot() {
+    return _relRoot;
+  }
```

Comment:
Isn't `QueryEnvironment` a global object used for all queries going via the `MultiStageBrokerRequestHandler`? There is a single `MultiStageBrokerRequestHandler` object. How will this work if there are multiple queries issues in parallel via the `MultiStageBrokerRequestHandler`?

Reply:
Good point. This wouldn't work. Let me think of an alternate way.